src/idl_parser.cpp

```diff
@@ -723,6 +725,10 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
     // This will cause an error if the root type of the nested flatbuffer
     // wasn't defined elsewhere.
     LookupCreateStruct(nested->constant);
+
+    // Keep a pointer to StructDef in attribute to simplify re-use later
+    auto nested_qualified_name = namespaces_.back()->GetFullyQualifiedName(nested->constant);
+    nested->type.struct_def = structs_.Lookup(nested_qualified_name);
```

Comment:
Yes, but you're creating a new kind of type that's a mix between `[ubyte]` and `my_nested_type` which is a bit weird. If storing it in `nested_flatbuffer` works, that is probably better.